### PR TITLE
remove kube-rbac-proxy in accordance with v1alpha4 migration

### LINKED ISSUE #1243: Remove kube-rbac-proxy
/kind cleanup

According to the v1alpha4 migration documentation kube-rbac-proxy should be removed: https://cluster-api.sigs.k8s.io/developer/providers/v1alpha3-to-v1alpha4

- cluster-api-provider-vsphere version: main

### PR DESCRIPTION
**Pull request by AartiJivrajani**
**What this PR does / why we need it**:
This PR removes kube-rbac-proxy in accordance with the v1alpha3 -> v1alpha4 migration. 
The steps are mentioned [here](https://cluster-api.sigs.k8s.io/developer/providers/v1alpha3-to-v1alpha4)

**Which issue(s) this PR fixes**
Fixes #1243 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```
NONE
```

### CODE REVIEW

File: config/rbac/kustomization.yaml [hunk 1]
```diff
@@ -4,7 +4,4 @@ resources:
 - role.yaml
 - role_binding.yaml
 - leader_election_role.yaml
-- leader_election_role_binding.yaml
-- auth_proxy_service.yaml
-- auth_proxy_role.yaml
-- auth_proxy_role_binding.yaml
+- leader_election_role_binding.yaml
```
Comment (gab-satchi) [inline config/rbac/kustomization.yaml:7]: missing new line

**Review by yastij — APPROVED** [review]
/approve
@gab-satchi for lgtm

**Pull request merged**